src/controllers/steps.controller.js

```diff
@@ -0,0 +1,115 @@
+const Step = require("../models/steps.model");
+const Course = require("../models/courses.model")
+
+// Create Step
+function create(req, res, next) {
+  Step.create({ ...req.body }, (err, newStep) => {
+    if (err) return next(err);
+    res.status(201).json({ message: "New step created", newStep });
+  });
+}
+
+// Add Step to course
+function addStep(req, res, next) {
+  const { stepId } = req.params;
+  const step = Step.findById(stepId);
+  
+  Course.findByIdAndUpdate(req.params.courseId, { $addToSet: { steps: [step] } }, (err, course) => {
+    if (err) return next();
+    if (!course) return next({
+      status: 404,
+      message: `Course id ${req.params.courseId} cannot be found`
+    });
+    course.save((err, savedCourse) => {
+      if (err) return next({
+        status: 400,
+        message: err
+      });
+      res.json({ message: "Step added to course", savedCourse });
+    });
+  });
+}
+
+// Remove step from course
+function removeStep(req, res, next) {
+  const { stepId } = req.params;
+  const step = Step.findById(stepId);
```

Comment:
callback applies here too

Reply:
Revision included in most recent commit "add stepExists and update addStep & removeStep"